File: app/src/main/java/com/smart_learn/presenter/activities/OpenLessonActivity.java
package com.smart_learn.presenter.activities;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import com.smart_learn.R;
import com.smart_learn.presenter.activities.dialogs.DialogButtonsActionCallback;
import com.smart_learn.presenter.activities.dialogs.DialogDismissCallback;
import com.smart_learn.presenter.activities.dialogs.LessonDialog;
import com.smart_learn.presenter.activities.dialogs.SettingsDialog;
import com.smart_learn.presenter.view_models.OpenLessonViewModel;

public class OpenLessonActivity extends AppCompatActivity {

    private OpenLessonViewModel openLessonViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_open_lesson);

        setSupportActionBar(findViewById(R.id.toolbarLessons));

        // set ViewModel
        openLessonViewModel = new ViewModelProvider(this).get(OpenLessonViewModel.class);

        // set observers
        openLessonViewModel.getLiveToastMessage().observe(this, new Observer<String>() {
            @Override
            public void onChanged(String s) {
               Toast.makeText(OpenLessonActivity.this, s, Toast.LENGTH_LONG).show();
            }
        });



        // get GUI components
        Button btnNewLesson = findViewById(R.id.btnNewLesson);
        Button btnOpenLesson = findViewById(R.id.btnOpenLesson);

        // set listeners
        btnNewLesson.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                showAddLessonDialog();
            }
        });

        btnOpenLesson.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                //startLessonRVActivity();
            }
        });

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.toolbar_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        if (item.getItemId() == R.id.btnSettings) {
            showSettingsDialog();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    public void startLessonRVActivity(){
        //Intent intent = new Intent(this, LessonRVActivity.class);
        //startActivity(intent);
    }

    private void showSettingsDialog(){
        DialogFragment dialogFragment = new SettingsDialog(true, R.string.settings, R.layout.dialog_settings);
        dialogFragment.show(getSupportFragmentManager(), "OpenLessonActivity");
    }

    private void showAddLessonDialog(){

        DialogFragment dialogFragment = new LessonDialog(true, R.string.add_lesson, R.layout.dialog_new_lesson,
                R.string.save, R.string.cancel, new DialogButtonsActionCallback() {

            @Override
            public void onPositiveButtonPressed(View view, DialogInterface dialog, int which) {
                EditText etLessonName = ((AlertDialog) dialog).findViewById(R.id.etLessonName);
                openLessonViewModel.processLessonDialog(etLessonName, new DialogDismissCallback() {
                    @Override
                    public void onDismiss() {
                        dialog.dismiss();
                    }
                });
            }

        });

        dialogFragment.show(getSupportFragmentManager(), "OpenLessonActivity");

    }

}

File: app/src/main/java/com/smart_learn/presenter/activities/OpenLessonActivity.kt
package com.smart_learn.presenter.activities

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import com.smart_learn.R
import com.smart_learn.general.showSettingsDialog
import com.smart_learn.services.activities.MainActivityService

class OpenLessonActivity : AppCompatActivity() {

    private lateinit var mainActivityService: MainActivityService

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_open_lesson)
        setSupportActionBar(findViewById(R.id.toolbarDictionaries))

        mainActivityService = MainActivityService(this)

    }


    /** Inflate the menu. This adds items to the action bar if it is present. */
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.toolbar_menu, menu)
        return true
    }


    /** Handle presses on the action bar menu items */
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.btnSettings -> {
                showSettingsDialog(this)
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }


    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }


    override fun onDestroy() {
        super.onDestroy()
        Log.d("[APPLICATION CLOSED]","Application was successfully closed")
    }


    /**
     * These functions are for activity management
     * */
    fun startDictionariesRVActivity(){
        val intent = Intent(this, DictionariesRVActivity::class.java)
        startActivity(intent)
    }

    fun getActivity(): Activity { return this }

}
